tests/e2e/compare/output/markdown.ts

```diff
@@ -1,80 +1,85 @@
 // From: https://raw.githubusercontent.com/callstack/reassure/main/packages/reassure-compare/src/output/markdown.ts
 import fs from 'node:fs/promises';
 import path from 'path';
-import _ from 'underscore';
+import type {Stats} from 'tests/e2e/measure/math';
 import * as Logger from '../../utils/logger';
+import type {Data, Entry} from './console';
 import * as format from './format';
 import markdownTable from './markdownTable';
 
 const tableHeader = ['Name', 'Duration'];
 
-const collapsibleSection = (title, content) => `<details>\n<summary>${title}</summary>\n\n${content}\n</details>\n\n`;
+const collapsibleSection = (title: string, content: string) => `<details>\n<summary>${title}</summary>\n\n${content}\n</details>\n\n`;
 
-const buildDurationDetails = (title, entry) => {
+const buildDurationDetails = (title: string, entry: Stats) => {
     const relativeStdev = entry.stdev / entry.mean;
 
-    return _.filter(
-        [
-            `**${title}**`,
-            `Mean: ${format.formatDuration(entry.mean)}`,
-            `Stdev: ${format.formatDuration(entry.stdev)} (${format.formatPercent(relativeStdev)})`,
-            entry.entries ? `Runs: ${entry.entries.join(' ')}` : '',
-        ],
-        Boolean,
-    ).join('<br/>');
+    return [
+        `**${title}**`,
+        `Mean: ${format.formatDuration(entry.mean)}`,
+        `Stdev: ${format.formatDuration(entry.stdev)} (${format.formatPercent(relativeStdev)})`,
+        entry.entries ? `Runs: ${entry.entries.join(' ')}` : '',
+    ]
+        .filter(Boolean)
+        .join('<br/>');
 };
 
-const buildDurationDetailsEntry = (entry) =>
-    _.filter(['baseline' in entry ? buildDurationDetails('Baseline', entry.baseline) : '', 'current' in entry ? buildDurationDetails('Current', entry.current) : ''], Boolean).join(
-        '<br/><br/>',
-    );
+const buildDurationDetailsEntry = (entry: Entry) =>
+    ['baseline' in entry ? buildDurationDetails('Baseline', entry.baseline) : '', 'current' in entry ? buildDurationDetails('Current', entry.current) : '']
+        .filter(Boolean)
+        .join('<br/><br/>');
+
+const formatEntryDuration = (entry: Entry): string => {
+    let formattedDuration = '';
 
-const formatEntryDuration = (entry) => {
     if ('baseline' in entry && 'current' in entry) {
-        return format.formatDurationDiffChange(entry);
+        formattedDuration = format.formatDurationDiffChange(entry);
```

Comment:
this is a bit more verbose, wondering why we needed this vs what we had before, did TS complain?

Reply:
@youssef-lr Yeah, TS complained before!